server/fleet/app.go

```diff
@@ -443,7 +443,17 @@ func (c *AppConfig) Copy() *AppConfig {
 	if c.Features.AdditionalQueries != nil {
 		aq := make(json.RawMessage, len(*c.Features.AdditionalQueries))
 		copy(aq, *c.Features.AdditionalQueries)
-		c.Features.AdditionalQueries = &aq
+		clone.Features.AdditionalQueries = &aq
+	}
+	if c.Features.DetailQueryOverrides != nil {
```

Comment:
This worries me a bit - it was missing from the appconfig's cloning when the field was added. It's very error-prone and easy to miss one (and line 446 is another worrying example, it was not assigning the copied value to the clone).

We have "WARNING" comments in the related structs, but that's obviously not infallible. Anyone has better ideas on how to ensure we don't miss such cases going forward? I'll create a new ticket to address this. We need to a) (ideally) ensure we do use a custom clone implementation (easy with an API change of the cache) and b) ensure we update the custom implementation each time we add new fields with pointers/slices/maps to those impacted structs. 

Only thing I can think of at the moment for b) would be to generate a (recursive) listing of existing fields in those cached data structures into a text file (a bit like [Go does for backwards API compatibility](https://github.com/golang/go/blob/master/api/go1.21.txt)) and a CI check that fails if it finds fields that are not listed in the text file (and have a make target to re-generate the file, but only do so once we're sure the new fields have been addressed if needed in the cloning).

Reply:
this was the cause of a bug indeed (I'm going to close https://github.com/fleetdm/fleet/pull/14291 in favor of this)

> Only thing I can think of at the moment for b) would be to generate a (recursive) listing of existing fields in those cached data structures into a text file (a bit like Go does for backwards API compatibility) and a CI check that fails if it finds fields that are not listed in the text file (and have a make target to re-generate the file, but only do so once we're sure the new fields have been addressed if needed in the cloning).

sounds great! I thought about a hardcoded slice + reflection, but a file sounds better as it can be dynamically generated more easily

Reply:
Ticket created: https://github.com/fleetdm/fleet/issues/14979